### PR TITLE
javascript binary search tree: Add Additional resources 

### LINKED ISSUE #25095: Improvement for Binary Search Tree project
### Describe your suggestion

Had some suggestions for videos to help students visualize and understand the different traversal methods ( inorder, preorder, postorder ) for the BST project.

Personally found these videos to be super helpful in
1. Visualizing how the tree will work step by step
2. Understanding recursion at a deeper level coupled with the visuals of step 1.

I think these videos by ygongcode would be helpful additions

Inorder Video: https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s&ab_channel=ygongcode

Postorder video: https://www.youtube.com/watch?v=4Xo-GtBiQN0&ab_channel=ygongcode

Pre-order video: https://www.youtube.com/watch?v=8xue-ZBlTKQ&ab_channel=ygongcode

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-binary-search-trees

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

Hello KT#0001

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by taladan**
This adds the links suggested by @LuvKatie in issue #25095

fixes: #25095

## Because

Add additional video links to aid understanding of BST traversal algorithms.

## This PR

- Add additional resources section to JS binary search tree lesson

## Issue

Closes #25095 

## Additional Information

None

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/computer_science/project_binary_search_trees.md [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
 
 1.  Build a `Tree` class / factory which accepts an array when initialized. The `Tree` class should have a `root` attribute which uses the return value of `buildTree` which you'll write next.
 
 1.  Write a `buildTree` function which takes an array of data (e.g. [1, 7, 4, 23, 8, 9, 4, 3, 5, 7, 9, 67, 6345, 324]) and turns it into a balanced binary tree full of `Node` objects appropriately placed (don't forget to sort and remove duplicates!). The `buildTree` function should return the level-0 root node.
 
     **Tip:** If you would like to visualize your binary search tree, here is a `prettyPrint()` function that will `console.log` your tree in a structured format. This function will expect to receive the root of your tree as the value for the `node` parameter.
 
-    ~~~javascript
-    const prettyPrint = (node, prefix = '', isLeft = true) => {
+    ```javascript
```
Comment (Lofty-Brambles) [inline javascript/computer_science/project_binary_search_trees.md:21]: `~~~` would be in accordance to TOP's style guide here.

File: javascript/computer_science/project_binary_search_trees.md [hunk 2]
```diff
@@ -66,3 +67,13 @@ Write a simple driver script that does the following:
...
 
 </div>
+
+### Additional Resources
+
+This section contains helpful links to related content. It isn't required, so consider it supplemental.
+
+If you need more help on the three types of binary search tree traversal algorithms, these video resources may aid you:
+
+- [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s) of a binary search tree by Yicheng Gong
```
Comment (Lofty-Brambles) [inline javascript/computer_science/project_binary_search_trees.md:77]: I like the way this is added here, but could you explain why and when a learner should watch these? Something in the lines of "this helps in further visualising how a call stack looks when traversing the tree" should be cool enough IMO.

**Review by Lofty-Brambles — COMMENTED** [review]
Just a couple of suggestions.

File: javascript/computer_science/project_binary_search_trees.md (hunk 1 above)
Comment (Lofty-Brambles) [reply javascript/computer_science/project_binary_search_trees.md:21]: @taladan - You can commit and push to your remote branch. It'll get reflected.

Comment (Lofty-Brambles) [reply javascript/computer_science/project_binary_search_trees.md:21]: And yes, VSCode's auto-save formatting defaults to backticks for fenced codeblocks.

**Comment by taladan** [comment]
Implemented Lofty's suggestions as they were good ones.

File: javascript/computer_science/project_binary_search_trees.md (hunk 2 above)
Comment (taladan) [reply javascript/computer_science/project_binary_search_trees.md:77]: This has been implemented

File: javascript/computer_science/project_binary_search_trees.md [hunk 3]
```diff
@@ -66,3 +67,13 @@ Write a simple driver script that does the following:
...
+
+### Additional Resources
+
+This section contains helpful links to related content. It isn't required, so consider it supplemental.
+
+If you need more help on the three types of binary search tree traversal algorithms, these video resources may aid you in visualizing how a call stack looks when traversing the tree:
+
+- [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s) of a binary search tree by Yicheng Gong
+- [Post-order Traversal Algorithm](https://www.youtube.com/watch?v=4Xo-GtBiQN0) of a binary search tree by Yicheng Gong
+- [Pre-order Traversal Algorithm](https://www.youtube.com/watch?v=8xue-ZBlTKQ&ab_channel=ygongcode) of a binary search tree by Yicheng Gong
```
Comment (rlmoser99) [inline javascript/computer_science/project_binary_search_trees.md:79]: This does not follow our layout style guide for Additional Resources. The explanation before 3 bullet points do not allow for additional bullet points at a later time. 
https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md

In addition, the 3 bullet points are repetitive, so I think we should try and condense it down to one bullet point that contains all 3 links. Something like:

```suggestion
- Yicheng Gong has some excellent videos that help visualize the call stack when traversing binary search trees: [Inorder Traversal Algorithm](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s), [Post-order Traversal Algorithm](https://www.youtube.com/watch?v=4Xo-GtBiQN0), and [Pre-order Traversal Algorithm](https://www.youtube.com/watch?v=8xue-ZBlTKQ). 
```

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thanks for taking the time to update this lesson. I have requested a few changes.

File: javascript/computer_science/project_binary_search_trees.md (hunk 3 above)
Comment (rlmoser99) [reply javascript/computer_science/project_binary_search_trees.md:79]: We might be able to condense the links even more, something like:
```suggestion
- Yicheng Gong has some excellent videos that help visualize the call stack when traversing binary search trees: [Inorder](https://www.youtube.com/watch?v=4_UDUj1j1KQ&t=1s), [Post-order](https://www.youtube.com/watch?v=4Xo-GtBiQN0), and [Pre-order](https://www.youtube.com/watch?v=8xue-ZBlTKQ) Traversal Algorithms.
```

**Review by rlmoser99 — APPROVED** [review]
Thank you for taking the time to make this change, and all the revisions!

**Pull request merged**